### DISCUSSION TITLE
ConfigMap mount error to /opt/airflow

### DISCUSSION THREAD

**Discussion by erosons** [apache/airflow#33788] [General]
My [values.txt](https://github.com/apache/airflow/files/12446907/values.txt)`

### Docker Image customizations

I rebuilt the airflow base image and use it to replace the current image but using the default image can still replay the scenario

### What happened

I created a configMap 
` >> kubectl create secret generic web-cert-and-key --from-file=/Users/s.eromonsei/Downloads/server.key   --from-file=/Users/s.eromonsei/Downloads/server.crt -n airflow-dev`

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: ssl-keys
data:
  server.crt: |
    -----BEGIN CERTIFICATE-----

    -----END CERTIFICATE-----
  server.key: |
    -----BEGIN RSA PRIVATE KEY-----
    
        -----END RSA PRIVATE KEY-----
```

 I am trying to mount the configMap  airflow webserver to the following path using vulomeMount=>  /opt/airflow  I add my configuration to extraVolumes  and extraMount, With the hope to reference the ssl  crt and  key in the mount path to the .cfg file for  as shown below
   ```
 web_server_ssl_key = /opt/airflow/server.crt
      web_server_ssl_cet = /opt/airflow/server.key
```
`Error: failed to create containerd task: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/var/lib/kubelet/pods/0c8d61a1-2bcb-448e-abe9-124b5c35dd1c/volume-subpaths/config/wait-for-airflow-migrations/0" to rootfs at "/opt/airflow/airflow.cfg": mount /var/lib/kubelet/pods/0c8d61a1-2bcb-448e-abe9-124b5c35dd1c/volume-subpaths/config/wait-for-airflow-migrations/0:/opt/airflow/airflow.cfg (via /proc/self/fd/6), flags: 0x5001: not a directory: unknown`

### What you think should happen instead

My expectation is the data point  server.crt and server.key will be available in the point , I can then pass to the airflow.cfg file 
   ```
 web_server_ssl_key = /opt/airflow/server.crt
      web_server_ssl_cet = /opt/airflow/server.key
```

### How to reproduce

```
>>> helm repo add apache-airflow https://airflow.apache.org
>>> helm repo update
>>> helm search repo airflow
>>> helm install airflow apache-airflow/airflow --namespace airflow --debug
 Get airflow values into a yaml file
   >>> helm show values apache-airflow/airflow > values.yaml
Created a configmap with two data two for ssl crt and key as shown above  and added to the ExtraVolume and ExtraMounts under the webserver
```

### Anything else

N/A

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

```

**Comment by potiuk** [comment]
You can't mount folder as /opt/airflow, because then airflow Helm will not be able to mount "airflow.cfg" file as secret in your mounted secret as folder (this is what you are observing)

**Reply by potiuk** [reply]
Converted to discussion if needed.

**Reply by erosons** [reply]
Is there any applicable alternative to map  those ssl crt and key to web_server_ssl_key ,web_server_ssl_cet . I tried hardcoding under the config/Webserver , iwas getting the error expected  ":"
 ```
 web_server_ssl_key:  |
    -----BEGIN CERTIFICATE-----

    -----END CERTIFICATE-----
  web_server_ssl_cet:  |
    -----BEGIN RSA PRIVATE KEY-----
    
        -----END RSA PRIVATE KEY-----
```

**Reply by potiuk** [reply]
You can mount them where you want and point to them via configuration:

Here are docs how you can change configuration in the chart  https://airflow.apache.org/docs/helm-chart/stable/airflow-configuration.html#configuring-airflow and here are those config options in the docs of airflow https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#web-server-ssl-cert

**Reply by erosons** [reply]
thank you for the docs. it appears I was missing some understand. I have set the ssl and crt as env.
reference them in my .cfg file  but there seems to  be an issue refeence them the pod


From my Values.yml file
  ```
webserver:
    enable_proxy_fix: 'True'
    base_url: 'https://airflow-ses-dev.ses.shell.com:8080'
    web_server_ssl_key: ${AIRFLOW__WEBSERVER__WEB_SERVER_SSL_KEY}
    web_server_ssl_cert: ${AIRFLOW__WEBSERVER__WEB_SERVER_SSL_CERT}
```

airflow.cfg in the pod,I was expecting it to reference the variables, rather  it calling the string as it is

<img width="875" alt="image" src="https://github.com/apache/airflow/assets/73761240/bea0ce77-6c9b-4b17-9e0d-599572d981c3">
